src/encoded/static/components/static-pages/placeholders/MdSortableTable.js

```diff
@@ -0,0 +1,646 @@
+'use strict';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import Markdown from 'markdown-to-jsx';
+import Draggable from 'react-draggable';
+
+import { ajax } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
+
+export class MdSortableTable extends React.PureComponent {
+
+    static propTypes = {
+        'mdFilePath' : PropTypes.string,
+        'content': PropTypes.string,
+        'defaultColWidths': PropTypes.array
+    };
+
+    constructor(props) {
+        super(props);
+        const { content, mdFilePath } = props;
+        this.state = {
+            data: content ? Utils.convertMarkdownTableToObject(content) : null,
+            loading: !content && (mdFilePath && typeof mdFilePath === 'string')
+        };
+    }
+
+    componentDidMount(){
+        this.loadMdFile();
+    }
+
+    loadMdFile() {
+        const { mdFilePath } = this.props;
+        const onFinishLoad = (data) => {
+            this.setState({ 'loading': false, data: data });
+        };
+
+        if (mdFilePath) {
+            window.fetch = window.fetch || ajax.fetchPolyfill; // Browser compatibility polyfill
+            window.fetch(mdFilePath)
+                .then(function (resp) {
+                    return resp.text();
+                })
+                .then(function (respText) {
+                    const data = Utils.convertMarkdownTableToObject(respText);
+                    onFinishLoad(data);
+                });
+        }
+    }
+
+    render() {
+        const { defaultColWidths } = this.props;
+        const { data, loading } = this.state;
+        if (!data || !Array.isArray(data) || data.length === 0) {
+            if (loading) {
+                return (
+                    <div className="text-center" style={{ paddingTop: 20, paddingBottom: 20, fontSize: '2rem', opacity: 0.5 }}>
+                        <i className="icon icon-fw icon-spin icon-circle-notch fas" />
+                    </div>
+                );
+            }
+            return null;
+        }
+
+        const columns = _.map(Object.keys(data[0]), function (item) {
+            return {
+                header: item,
+                key: item,
+                sortable: true,
+                //defaultSorting: 'ASC',
+                descSortFunction: CustomSorter.desc,
+                ascSortFunction: CustomSorter.asc,
+                //render: (id) => { return (<a href={'user/' + id}>{id}</a>); }
+            };
+        });
+
+        return (
+            <SortableTable {...{ data, columns, defaultColWidths }} />
+        );
+    }
+}
+
+const CustomSorter = {
+    desc: (data, key) => {
```

Comment:
(Low importance) Many of these arrow functions that are properties of a static object variable could be regular pure functions as no dependence on "this" variable (or parent scope) is needed. More of code style preference than anything (arguably helps identify functions with no side effects ("function(..){..}") vs those which have side effects or dependencies on a parent scope (arrow funcs or methods which use "this" / set or use state / call other methods of "this" / etc)